text/0033-logging.md

```diff
@@ -0,0 +1,204 @@
+|-------|-------------------------------------------------------------------------------------------------------------------|
+| ERROR | Everything related to an error. A non blocking error has occured.                                                 |
+| WARN  | Used to warn about an exceptional non-blocking event occurring.                                                   |
+| INFO  | Default Log Level. It displays high level informations of events occuring in the search engine.                   |
+| DEBUG | Used for debugging and development purposes.  More verbose than INFO.                                             |
+| TRACE | Display everything happening at engine level. Can be useful for rust developers dealing with complex issues       |
+
+
+##### Log Format
+
+```
+[2021-03-02T20:33:09Z INFO actix_web::middleware::logger] 172.17.0.1:57220 "POST /indexes/indexUID/documents HTTP/1.1" 202 14 "-" "PostmanRuntime/7.26.10" 0.023529
+```
+
+###### Mandatory log format part. E.g [TIME_FORMAT LOG_LEVEL MODULE] part.
+
+- Time when the request was started to process (in rfc3339 format)
+- Log levels are  `ERROR`, `WARN`, `INFO`, `DEBUG`, `TRACE`.
+- The module part gives information about the module that records the log.
+
+###### HTTP Call
+
+Transplant uses `actix_web::middleware::logger` to record information about the API endpoints that receive calls.
+
+Given
+```172.17.0.1:57220 "POST /indexes/indexUID/documents HTTP/1.1" 202 14 "-" "PostmanRuntime/7.26.10" 0.023529```
+
+- Peer IP address (or IP address of reverse proxy if used)
+- First line of request (Example: GET /test HTTP/1.1)
+- Response status code
+- Size of response in bytes, including HTTP headers
+- User-Agent
+- Time taken to serve the request, in seconds to 6 decimal places
+
+> At DEBUG log level, the search endpoint must log the request body and the response body.
+
+### V. Impact on Documentation
+
+The documentation only mentions the logging behavior for the `development` env on the `MEILI_ENV` part.
+
+We should explain how to specify the logging level using the `MEILI_LOG_LEVEL` environment variable and display the logging level table as information in a dedicated section. It should also mention the usage of the cli flag `--log-level`.
+
+### VI. Impact on SDKs
+N/A
+
+## 2. Technical Aspects
+N/A
+
+## 3. Future Possibilities
+
+- Store logs on filesystem (give us future possibilites of rolling strategy). We will keep an eye on https://roadmap.meilisearch.com/c/81-specify-log-path, Github issues and, Slack Community messages. Keep in mind that it is possible to send logs to files using `syslog` or `systemd` journalctl.
+- Develop an API endpoint to search for logged events and configure the logging policy for the instance (SaaS feature in mind).
+- Add syntaxic sugar helper flag like `-v, -vv, -vvv` that can be translated to a `MEILI_LOG_LEVEL` value.
+
```

Comment:
```suggestion
- Add syntactic sugar helper flag like `-v, -vv, -vvv` that can be translated to a `MEILI_LOG_LEVEL` value.
- Use the rust clone of `log4j2`, [`log4rs`](https://docs.rs/log4rs/1.0.0/log4rs/) instead of `env_logger` to provide the same kind of options that Elasticsearch propose.
```

From what I understand, from a user perspective, Elasticsearch has the best logging system, so I guess we should consider using `log4rs` to be as good as them.

Reply:
Indeed, this is something that could be added to allow users to have a more versatile logging policy, I imagine it will become very useful when we have high availability and logs will have to be collected remotely. The rolling log method could be very useful at that time.